src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2327,6 +2344,12 @@ void Session::receive_download_message(const SyncProgress& progress, std::uint_f
                                        DownloadBatchState batch_state, int64_t query_version,
                                        const ReceivedChangesets& received_changesets)
 {
+    // Ignore download messages when the client detects an error. This is to prevent transforming the same bad
+    // changeset over and over again.
+    if (m_client_error) {
```

Comment:
can we do this after the logger call to `Received: DOWNLOAD` and log a debug statement that we're ignoring the download because we've sent an error message?